pyro/ops/einsum/__init__.py

```diff
@@ -35,4 +39,23 @@ def contract(equation, *operands, **kwargs):
         return expr(*operands, backend=backend, out=out)
 
 
-__all__ = ['contract', 'shared_intermediates']
+@contextlib.contextmanager
+def cached_paths(filename):
+    """
+    Context manager to load and save cached paths. This is most useful outside
+    of the training loop, so that optimized einsum paths can be saved across
+    training runs. Note that this saves even after user interrupt with CTRL-C.
+
+    :param str filename: path to a pickle file where the cache will be stored
+    """
+    if os.path.exists(filename):
+        with open(filename, 'rb') as f:
+            _PATH_CACHE.update(pickle.load(f))
```

Comment:
This is neat! I am guessing that this is particularly for the `einsum` computation and not the other tensor computations keyed in by `id`?

Reply:
That's right, this is purely for caching einsum paths (the optimized operation schedules, not actual tensor comptuations; no tensors will be cached)